### PR TITLE
llvmPackages_git: 20.0.0-git-2024-09-16 -> 20.0.0-git-2024-09-22

### PR DESCRIPTION
**Pull request by RossComputerGuy**
## Description of changes

Upstream PR for Clang fix: https://github.com/llvm/llvm-project/pull/109601

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by alyssais — APPROVED** [review]

**Comment by RossComputerGuy** [comment]
Replaced the patch I made with a proper fix for the vt_gen issue, https://github.com/llvm/llvm-project/pull/109817

File: pkgs/development/compilers/llvm/common/default.nix [hunk 1]
```diff
@@ -483,7 +483,18 @@ let
...
             })
-          ];
+          ]
+          ++
+            lib.optional (lib.versions.major metadata.release_version == "20")
+              # Fix for vt_gen, remove during next git bump
+              # https://github.com/llvm/llvm-project/pull/109817
+              (
+                fetchpatch {
+                  url = "https://github.com/llvm/llvm-project/commit/af97bd22fd75eb84f27b0106ae7a7080dd06669b.patch";
```
Comment (alyssais) [inline pkgs/development/compilers/llvm/common/default.nix:493]: > This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.

This probably means this isn't a stable URL — I know this is only going to be around for a week, but it's still nice for people to be able to build old Nixpkgs in the future.

**Review by alyssais — CHANGES_REQUESTED** [review]

File: pkgs/development/compilers/llvm/common/default.nix (hunk 1 above)
Comment (RossComputerGuy) [reply pkgs/development/compilers/llvm/common/default.nix:493]: It's the commit hash of that PR which was merged upstream, idk how it wouldn't be a part of upstream after being merged.

Comment (alyssais) [reply pkgs/development/compilers/llvm/common/default.nix:493]: It looks like it was squash merged.

Comment (RossComputerGuy) [reply pkgs/development/compilers/llvm/common/default.nix:493]: So we should use https://github.com/llvm/llvm-project/commit/cace9869775a185793122f845d81a5ff46f15728 instead?

Comment (alyssais) [reply pkgs/development/compilers/llvm/common/default.nix:493]: Yeah.

Comment (alyssais) [reply pkgs/development/compilers/llvm/common/default.nix:493]: Or just update to latest git.

Comment (RossComputerGuy) [reply pkgs/development/compilers/llvm/common/default.nix]: I updated the patch, didn't feel like bumping to a latest git and trying to get this done fully before work. Also nice to have the branch and commit match lol.

**Review by alyssais — APPROVED** [review]

File: pkgs/development/compilers/llvm/common/default.nix (hunk 1 above)
Comment (alyssais) [reply pkgs/development/compilers/llvm/common/default.nix]: That's fine.

**Pull request merged**